sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -775,6 +1210,162 @@ Exceptions:
 * Throws synchronously with error code `invalid` if the path is invalid or
 the file extension is not supported or if the write operation failed.
 
+|
+[source,c++]
+----
+std::vector<node> get_nodes() const;
+----
+|Returns a list of all the nodes present in the graph in the order that they
+were added.
+
+|
+[source,c++]
+----
+std::vector<node> get_root_nodes() const;
+----
+|Returns a list of all nodes in the graph which have no dependencies in the
+order they were added to the graph.
+
+|===
+
+Table {counter: tableNumber}. Member functions of the `command_graph` class for
+graph update.
+[cols="2a,a"]
+|===
+|Member function|Description
+
+|
+[source,c++]
+----
+void update_dynamic_parameter(
+    void* newValue,
+    dynamic_parameter<dynamic_parameter_type::usm> dynamicParam);
+----
+
+| Updates USM parameters in nodes associated with `dynamicParam` to a new value.
+This new value will not affect any prior submissions of this graph and will
+take affect only for future submissions. See <<executable-graph-update,
+Executable Graph Update>> for more information about updating node parameters.
+
+Preconditions:
+
+* This member function is only available when the `command_graph` state is
+  `graph_state::executable`.
+
+Parameters:
+
+* `newValue` - The new value for the parameter being updated.
+
+* `dynamicParam` - The object which represents the USM node parameters to be
+  updated.
+
+Exceptions:
+
+* Throws synchronously with error code `invalid` if
+  `property::graph::updatable_graph` was not set when the executable graph
+  was created.
+
+* Throws synchronously with error code `invalid` if `newValue` is not a pointer
+  to a valid USM allocation.
+
+|
+[source,c++]
+----
+template <typename T>
+void update_dynamic_parameter(
+    T newValue,
+    dynamic_parameter<dynamic_parameter_type::buffer> dynamicParam);
+----
+
+| Updates buffer parameters in nodes associated with `dynamicParam` to a new value.
+This new value will not affect any prior submissions of this graph and will
+take affect only for future submissions. See <<executable-graph-update,
+Executable Graph Update>> for more information about updating node parameters.
+
+Preconditions:
+
+* This member function is only available when the `command_graph` state is
+  `graph_state::executable`.
+
+Parameters:
+
+* `newValue` - The new value for the parameter being updated.
+
+* `dynamicParam` - The object which represents the buffer node parameters to be
+  updated.
+
+Exceptions:
+
+* Throws synchronously with error code `invalid` if
+  `property::graph::updatable_graph` was not set when the executable graph
+  was created.
+
+* Throws with error code `invalid` if `newValue` is not a valid buffer accessor.
+
+|
+[source,c++]
+----
+template <typename T>
+void update_dynamic_parameter(
```

Comment:
Do we need to have a version of this method taking argument index ? Similar to the one used for registration of dynamic parameters

Reply:
I don't think so, the registration step is specifically for making this association between the dynamic_parameter object and the argument indices of the nodes to update. Once that association is made, we can update here without the user needing to pass those indices again. Indeed it might, complicate things as the you could then pass conflicting argument indexes to the ones registered with.

We've not added an example snippet to the spec before now to avoid making the spec too large, but I think this feature is complicated/important enough to warrant it. So i'll add something to our examples section to should how the API would be used to update arguments based on index.